### PR TITLE
Hackaton - Added bash script and template files for feature creation

### PR DESCRIPTION
**Pull request by Pelkar**
During the hackaton I created a bash script for generating new feature files.

Instruction:
* run ./glia_acmu.sh

* Enter feature name for example NewFeature. This will be used as the base for class naming for example NewFeatureActivity, NewFeatureManager etc.

* Enter package name from widgetssdk, both "/" and "." can be used for package separators. Example: new/feature

* Select feature set to create:
"a for *Activity, *View and *Contract"
"c for *Controller and *Contract"
"m for *Manager, *Repository and a basic feature interface"
"u for *UseCase"
The order does not matter

This is for review only. Please do not merge into master.

### CODE REVIEW

File: widgetssdk/templates/activity_template.txt [hunk 1]
```diff
@@ -0,0 +1,16 @@
+package com.glia.widgets.${packageReplacement}
```
Comment (AndriiHorishniiMOC) [inline widgetssdk/templates/activity_template.txt:1]: Maybe `.txt` is not a good file extension for this file?

**Comment by Pelkar** [comment]
> Honestly, I don't understand why this is necessary. In my experience, such utilities become obsolete quickly. Anyway, the script looks good.

I wanted to learn more about bash shell and took a goal for myself. As I did it during company time, then I felt that I should show this to our team. I am not expecting it to be merged to master in our project, but instead this is for evaluation purposes. If someone wants to make a copy/improve/redo, go ahead, but this should not be included in the public repository. Should have been more clear in the description.

File: glia_acmu.sh [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+           .--./)  |   ||__|
+          /,--\ \  |   |.--.
+         | |  | |  |   ||  |  _____
+          \ \_/ /  |   ||  | /,--.'\\
+          /(--'    |   ||  |/ |   \ |
+          \ '---.  |   ||  |-- __ | |
+           /-----.\|   ||__| / .--| |
+          ||     |||___|    / /   | |_
+          \'. __/ /         \ \._,\ '/
+           \_____/           \___/ \/   ACMU"
```
Comment (andrews-moc) [inline glia_acmu.sh:156]: I like it. 🙂 Creative approach. 👍

**Review by andrews-moc — APPROVED** [review]
Thank you for your initiative. 👍 
I think even if we don't use it, having templates for components is good in terms of project architecture agreements. 👍

**Comment by andrews-moc** [comment]
We can discuss if we want to merge it or not within the Android team.

File: widgetssdk/templates/view_template.txt [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+    }
+
+    override fun onLayout(changed: Boolean, left: Int, top: Int, right: Int, bottom: Int) {
+        super.onLayout(changed, left, top, right, bottom)
+    }
+
+    override fun setController(controller: ${nameReplacement}Contract.Controller?) {
+            TODO("Not yet implemented")
+        }
```
Comment (yurii-glia) [inline widgetssdk/templates/view_template.txt:47]: looks like incorrect alignment, is it?

**Review by yurii-glia — COMMENTED** [review]
I heard some time ago that Kotlin by default has a visibility level `public` so should we use `internal` for generated structures?

**Review by DavDo — APPROVED** [review]

**Review by AndriiHorishniiMOC — APPROVED** [review]

**Pull request closed without merging**